homeassistant/components/bmw_connected_drive.py

```diff
@@ -0,0 +1,116 @@
+"""
+Reads vehicle status from BMW connected drive portal.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/bmw_connected_drive/
+"""
+import logging
+import datetime
+
+import voluptuous as vol
+from homeassistant.helpers import discovery
+from homeassistant.helpers.event import track_utc_time_change
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_USERNAME, CONF_PASSWORD
+)
+
+REQUIREMENTS = ['bimmer_connected==0.3.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'bmw_connected_drive'
+CONF_VALUES = 'values'
+CONF_COUNTRY = 'country'
+
+LENGTH_ATTRIBUTES = [
+    'remaining_range_fuel',
+    'mileage',
+    ]
+
+VAILD_ATTRIBUTES = LENGTH_ATTRIBUTES + [
```

Comment:
Typo, same as before.

Reply:
Ah, yes was a copy an past error. I removed the duplicated code.